### PR TITLE
chore: adr

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
We need ADRs so decisions made are clear

## Solution
Use the `adr-tools` cli command to generate folder and also, generate new ADRs.

write out existing decisions into our ADRs.

### CODE REVIEW

File: docs/studio/0003-permissions.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# 3. permissions
```
Comment (harishv7) [inline docs/studio/0003-permissions.md:1]: super nit can we capitalise the headings? 3. Permissions

File: docs/studio/0004-database-schema.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# 4. database schema
```
Comment (harishv7) [inline docs/studio/0004-database-schema.md:1]: super nit : capitalise Database Schema

File: docs/studio/0002-use-next.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# 2. Use next
```
Comment (harishv7) [inline docs/studio/0002-use-next.md:1]: Next.js

File: docs/studio/0002-use-next.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+Accepted
+
+## Context
+
+Isomer is undergoing a rebuild from the original application that was on express for backend + react/chakra for frontend
+
+## Decision
+
+Isomer is going to use starter-kit (prisma, next, kysely) to build our application
```
Comment (harishv7) [inline docs/studio/0002-use-next.md:15]: Might be worthwhile to mention this Next.js pages router. We also made a conscious decision not to go with RSC for reasons I have outlined [here](https://opengovproducts.slack.com/archives/CK68JNFHR/p1716775203875659)

File: docs/studio/0003-permissions.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+1. publishing of pages
+2. editing of pages
+3. adding/deletion of users
+
+Number 1/2 will only be enforced at the root folders of the site, and all sub-folders/pages will instead inherit their permissions from their root-level folder.
+For example, the site has the structure (A -> B -> C), such that A is the root level folder. Both B and C will not have permissions tied to them and will instead use the permissions linked to A.
+
+## Decision
+
+We will use role-based access control for our permissions model, and we will use [casl](https://casl.js.org/v6/en/) for the implementation of the library
```
Comment (harishv7) [inline docs/studio/0003-permissions.md:22]: We should mention that there are 2 modes of permissions being supported. By default all new sites get a site-wide RBAC for publisher, editor, admin

On a opt-in basis we will allow for granular resource-level based perms on the root level folders

File: docs/studio/0004-database-schema.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+Accepted
+
+## Context
+
+We need to implement a database schema to store both our pages and folders in our db
+
+## Decision
+
+1. We will combine both page/folders into 1 schema and use a json blob to represent the content of the page instead of separating.
```
Comment (harishv7) [inline docs/studio/0004-database-schema.md:15]: worth mentioning its a JSONB column in postgres

this also allows us to query within the json using query operators and perform indexing etc

**Comment by harishv7** [comment]
@seaerchin should we store these in the codebase itself or have a dedicated Notion?

**Comment by seaerchin** [comment]
@karrui @harishv7 - i'll mkae changes here first then you can copy to notion later?

File: docs/studio/0002-use-next.md (hunk 4 above)
Comment (seaerchin) [reply docs/studio/0002-use-next.md:15]: added a line

File: docs/studio/0003-permissions.md (hunk 5 above)
Comment (seaerchin) [reply docs/studio/0003-permissions.md:22]: good point, will do

**Comment by harishv7** [comment]
> @karrui @harishv7 - i'll mkae changes here first then you can copy to notion later?

@seaerchin okay sure. don't need to merge the PR, just make the edits and I will copy over

**Pull request closed without merging**